apps/api/src/lib/coscrad-query-language/aql/compile-aql-filter-block.ts

```diff
@@ -129,19 +132,93 @@ const compileSimpleFilterCondition = (
             );
         }
 
+        const {
+            expression: fieldRef,
+            nestedFieldNames,
+            isArray,
+        } = buildFieldRef(docRef, field, startingArgIndex);
+
+        if (params.length === 2) {
+            const languageCode = params[1];
+
+            /**
+             * If the text is empty, we return all resources who have a
+             * multilingual text item in this language for the given field.
+             */
+            if (searchText === '') {
+                if (isArray) {
+                    const filterStatement = `contains(${fieldRef}.items[*].languaugeCode,@args[${
+                        startingArgIndex + nestedFieldNames.length
+                    }])`;
+
+                    return {
+                        filterStatement,
+                        bindVars: { args: [...nestedFieldNames, languageCode] },
+                    };
+                }
+
+                const letVarName = `matches_${++varCount}`;
+
+                const letStatement = `
+                let ${letVarName} = (
+                    for i in ${fieldRef}.items
+                    filter i.languageCode == @args[${
+                        startingArgIndex + nestedFieldNames.length + 1
+                    }]
+                    limit 1
+                    return "match"
+                )
+                `;
+
+                const statement = `length(${letVarName}) > 0`;
+
+                return {
+                    letStatement,
+                    filterStatement: statement,
+                    bindVars: { args: [...nestedFieldNames, searchText, languageCode] },
+                };
+            }
+
+            /**
+             * We need to ensure that this is globally unique, otherwise
```

Comment:
There are two similar issues.
1. We need to ensure that the identifiers for `let` vars are globally unique.
2. We need to ensure that the identifiers for all bind vars are unique.
In the first pass, I used an array of `args` for the bind vars, but this lead to difficult to read queries. We may just want to use the `bindVar_{i}` approach. This context needs to be kept around as we recursively build the query. There's a chance that tagged templates can deliver a better DX for writing these queries.